File: EmployeePayrollWeb/src/app/classes/salary.ts
export interface ISalary {
    empName: string
    empContactNo: string
    employeeId: number
    salaryDate: string
    salaryId: number
    totalAdvance: number
    presentDays: number
    salaryAmount: number
}
export class salary {
    salaryId: number = 0;
    employeeId: number = 0;
    salaryDate?: Date = undefined;
    totalAdvance: number = 0;
    presentDays: number = 0;
    salaryAmount: number = 0;
  }

File: EmployeePayroll.API/Models/Salary.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EmployeePayroll.API.Models
{
    [Table("Salary")]
    public class Salary
    {
        [Key]
        public int SalaryId { get; set; }
        public int? EmployeeId { get; set; }
        public DateTime? SalaryDate { get; set; }
        [Precision(18, 2)]
        public decimal TotalAdvance { get; set; } = 0;
        public int? PresentDays { get; set; }
        [Precision(18, 2)]
        public decimal SalaryAmount { get; set; } = 0;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
